File: keysAndRooms/keysAndRooms.py
from typing import List
from queue import Queue

class Solution:
    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        visited = set()
        toVisit = Queue()
        toVisit.put(0)
        while not toVisit.empty():
            room = toVisit.get()
            visited.add(room)
            for key in rooms[room]:
                if key not in visited:
                    toVisit.put(key)
        return (len(visited) == len(rooms))

File: keysAndRooms/keysAndRooms.cs
public class Solution {
    public bool CanVisitAllRooms(IList<IList<int>> rooms) {
        HashSet<int> visited = new HashSet<int>();
        Queue<int> toVisit = new Queue<int>();
        toVisit.Enqueue(0);
        while(toVisit.Count != 0)
        {
            int room = toVisit.Dequeue();
            visited.Add(room);
            foreach(int key in rooms[room])
            {
                if(!visited.Contains(key))
                {
                    toVisit.Enqueue(key);
                }
            }
            
        }
        
        return (visited.Count == rooms.Count);
    }
}